tools/example-verification-signing/main.go

```diff
@@ -0,0 +1,162 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This package implements a sample server that implements the public health authority
+// verification protocol: https://github.com/google/exposure-notifications-server/blob/master/docs/design/verification_protocol.md
+//
+// To call this server using curl:
+// curl -d '{"verificationCode":"fakeCode","tekhmac":"replace w/ tek hmac"}' -H "Content-Type: application/json" -X POST http://localhost:8080/
+package main
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/asn1"
+	"fmt"
+	"log"
+	"math/big"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/google/exposure-notifications-server/internal/setup"
+	"github.com/google/exposure-notifications-server/internal/signing"
+	"github.com/google/exposure-notifications-server/pkg/api/v1alpha1"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/gin-gonic/gin"
+)
+
+// VerifyRequest is the JSON structure the server accepts in order to issue a certificate.
+type VerifyRequest struct {
+	VerificationCode string `json:"verificationCode"`
+	HMAC             string `json:"tekhmac"`
+}
+
+// VerifyResponse is the response API type.
+type VerifyResponse struct {
+	Error                   string `json:"error"`
+	VerificationCertificate string `json:"verificationCertificate"`
+}
+
+type config struct {
+	KeyManager signing.Config
+
+	SigningKey string `env:"SIGNING_KEY,required"`
+	KeyVersion string `env:"KEY_VERSION, default=1"`
+
+	// Standard claims for the certificate.
+	Audience      string        `env:"AUDIENCE, default=exposure-notifications-service"`
+	Issuer        string        `env:"ISSUER, default=Department of Health"`
+	ValidDuration time.Duration `env:"VALID_DURATION, default=5m"`
+}
+
+func (c *config) KeyManagerConfig() *signing.Config {
+	return &c.KeyManager
+}
+
+func main() {
+	ctx := context.Background()
+	var cfg config
+	env, err := setup.Setup(ctx, &cfg)
+	if err != nil {
+		log.Fatalf("setup.Setup: %v", err)
+	}
+	defer env.Close(ctx)
+
+	router := gin.Default()
+	signer, err := env.GetSignerForKey(ctx, cfg.SigningKey)
+	if err != nil {
+		log.Fatalf("unable to retrieve signing key %v, error: %v", cfg.SigningKey, err)
+	}
+
+	router.POST("/", func(c *gin.Context) {
+		// Parse the VerifyRequest
+		var request VerifyRequest
+		var response VerifyResponse
+		if err := c.ShouldBindJSON(&request); err != nil {
+			response.Error = err.Error()
+			c.JSON(http.StatusBadRequest, response)
+			return
+		}
+
+		now := time.Now().UTC()
+
+		// Normally - you would verify the verificationCode against a database and optionally
+		// assign transmission risk overrides.
+
+		// Here - we simply sign the claims and assum the verificationCode is valid.
+
+		// Build a JWT that contains the Standard and Extended claims as defined in
+		// pkg/api/v1alpha1/verification_types.go
+		claims := v1alpha1.NewVerificationClaims()
+		claims.PHAClaims["testkit"] = "55-HH-A7"
```

Comment:
does `testkit` have any significance or is this just part of the example, a 'test claim'?

Reply:
added comment.